android/src/main/java/flutter/overlay/window/flutter_overlay_window/OverlayService.java

```diff
@@ -146,7 +141,7 @@ public int onStartCommand(Intent intent, int flags, int startId) {
         params.gravity = Gravity.TOP | Gravity.LEFT;
         flutterView.setOnTouchListener(this);
         windowManager.addView(flutterView, params);
-        return START_STICKY;
+        return START_NOT_STICKY;
```

Comment:
This is the important change which prevents Android from restarting the service.